### PR TITLE
[Service Modal Routes 2a of 3] DCOS-12423: Show marathon errors in service config display 

### PR DESCRIPTION
**Pull request by mlunoe**
~~Service Modal Routes 1 of 3: https://github.com/dcos/dcos-ui/pull/1828~~ Merged!
Service Modal Routes 2a of 3: https://github.com/dcos/dcos-ui/pull/1853
Service Modal Routes 2b of 3: https://github.com/dcos/dcos-ui/pull/1871

This PR ensures that we do display marathon errors on the service configuration page, when they emerge.

~~Diff for review can be found here: https://github.com/dcos/dcos-ui/compare/mlunoe/feature/DCOS-12423-add-routes-for-the-new-create-service-modal...mlunoe/feature/DCOS-12423-add-routes-for-the-new-create-service-modal-2?expand=1~~ Merged!

### CODE REVIEW

**Comment by mlunoe** [comment]
Run unit tests

**Comment by mlunoe** [comment]
Run unit tests

File: plugins/services/src/js/containers/service-configuration/ServiceConfiguration.js [hunk 1]
```diff
@@ -64,11 +67,9 @@ class ServiceConfiguration extends mixin(StoreMixin) {
 
     const serviceConfiguration = service.getVersions().get(selectedVersionID);
 
-    onEditClick(service,
-      ServiceUtil.getDefinitionFromSpec(
-          new ApplicationSpec(serviceConfiguration)
-      )
-    );
+    onEditClick(service, getDefinitionFromSpec(
+        new ApplicationSpec(serviceConfiguration)
```
Comment (jfurrow) [inline plugins/services/src/js/containers/service-configuration/ServiceConfiguration.js]: #nit This indentation seems weird but if the linter is happy, I'm happy!

**Comment by jfurrow** [comment]
LGTM :shipit:

File: plugins/services/src/js/containers/service-configuration/ServiceConfiguration.js (hunk 1 above)
Comment (mlunoe) [inline plugins/services/src/js/containers/service-configuration/ServiceConfiguration.js]: Done!

**Comment by mlunoe** [comment]
Run integration tests

**Comment by mlunoe** [comment]
Run integration tests

File: plugins/services/src/js/components/modals/NewCreateServiceModal.js [hunk 2]
```diff
@@ -442,30 +442,13 @@ class NewCreateServiceModal extends Component {
...
-          let messages = item.errors;
-
-          if (existingMessages) {
-            messages = messages.concat(existingMessages);
-          }
-
-          errorsMap.set(item.path, messages);
-        });
-      }
+    const errorsMap = getErrorsMap(errors);
```
Comment (nLight) [inline plugins/services/src/js/components/modals/NewCreateServiceModal.js]: Is it something that conflicts with your work @wavesoft ?

Comment (mlunoe) [inline plugins/services/src/js/components/modals/NewCreateServiceModal.js]: It was, I have updated the PR now!

**Comment by mlunoe** [comment]
Ok, rebased this guy! @MatApple would ❤️ some 👀 on this :one: so we can make sure to display marathon errors if they are present!

**Comment by mlunoe** [comment]
This is btw. related to this issue I discovered: https://github.com/dcos/dcos-ui/pull/1830/files#r102314678

**Comment by mlunoe** [comment]
Run integration tests

File: plugins/services/src/js/components/modals/NewCreateServiceModalForm.js [hunk 3]
```diff
@@ -392,20 +358,27 @@ class NewCreateServiceModalForm extends Component {
...
   getSectionContent(data, errorMap) {
-    const errorsAlertComponent = this.getErrorsAlertComponent();
+    const {showAllErrors} = this.props;
+    const errors = this.getErrors();
 
     if (this.state.isPod) {
       return [
         <TabView id="networking" key="multinetworking">
-          {errorsAlertComponent}
+          <ErrorsAlert
```
Comment (MatApple) [inline plugins/services/src/js/components/modals/NewCreateServiceModalForm.js]: I wonder if this should be added to our `<TabView/>` component and we pass the props to it? Could you ever see a case where we might want to configure where the `<ErrorsAlert/>` component is placed?

File: plugins/services/src/js/containers/service-configuration/ServiceConfigurationContainer.js [hunk 4]
```diff
@@ -32,14 +32,18 @@ class ServiceConfigurationContainer extends React.Component {
...
+        onClearError={onClearError}
+        errors={errors}
         onEditClick={onEditClick}
         service={service} />
     );
   }
 }
 
 ServiceConfigurationContainer.propTypes = {
+  onClearError: React.PropTypes.func,
```
Comment (MatApple) [inline plugins/services/src/js/containers/service-configuration/ServiceConfigurationContainer.js]: Should we add some defaults for these `propTypes`?

File: plugins/services/src/js/containers/service-detail/ServiceDetail.js [hunk 5]
```diff
@@ -236,6 +245,8 @@ ServiceDetail.contextTypes = {
 
 ServiceDetail.propTypes = {
   actions: PropTypes.object.isRequired,
+  clearError: PropTypes.func,
```
Comment (MatApple) [inline plugins/services/src/js/containers/service-detail/ServiceDetail.js]: Lets add a default `propType` for this seeing we call it blindly.

File: src/js/components/ErrorsAlert.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import Alert from './Alert';
+import {getUnanchoredErrorMessage} from '../utils/ErrorMessageUtil';
+
+const ErrorsAlert = function (props) {
+  const {errors, showAllErrors, pathMapping} = props;
+  let showErrors = errors;
+
+  if (!showAllErrors) {
+    showErrors = showErrors.filter(function (error) {
+      return error.path.length === 0;
```
Comment (MatApple) [inline src/js/components/ErrorsAlert.js]: Perhaps we could get a comment on the `propType` at the bottom to tell us what `showAllErrors` does. Is it just showing high level errors?

File: plugins/services/src/js/components/modals/NewCreateServiceModalForm.js (hunk 3 above)
Comment (mlunoe) [inline plugins/services/src/js/components/modals/NewCreateServiceModalForm.js]: Yes, I do see places where we could use `Tabs` without it being in a form. I like the ability to compose. It is easy to understand what is going on and is in line with how we are going about creating components now.

File: src/js/components/ErrorsAlert.js (hunk 6 above)
Comment (mlunoe) [inline src/js/components/ErrorsAlert.js]: Renamed!

File: plugins/services/src/js/containers/service-detail/ServiceDetail.js (hunk 5 above)
Comment (mlunoe) [inline plugins/services/src/js/containers/service-detail/ServiceDetail.js]: Updated prop types altogether. They needed a little love :)

File: plugins/services/src/js/containers/service-configuration/ServiceConfigurationContainer.js (hunk 4 above)
Comment (mlunoe) [inline plugins/services/src/js/containers/service-configuration/ServiceConfigurationContainer.js]: Updated prop types altogether. They needed a little love :)

**Comment by MatApple** [comment]
@mlunoe - integration test is failing:

164 passing (9m)
  24 pending
  1 failing

  1) Service Form Modal Picker should have three options to choose from:
     CypressError: Timed out retrying: expected [ Array(4) ] to deeply equal [ Array(3) ]
      at Object.cypressErr (http://localhost:4200/__cypress/static/js/cypress.js:5719:15)
      at Object.throwErr (http://localhost:4200/__cypress/static/js/cypress.js:5684:22)
      at Object.throwErrByPath (http://localhost:4200/__cypress/static/js/cypress.js:5711:21)
      at $Cy._retry (http://localhost:4200/__cypress/static/js/cypress.js:6654:26)
      at http://localhost:4200/__cypress/static/js/cypress.js:8920:28
      at tryCatcher (http://localhost:4200/__cypress/static/js/vendor.js:46486:31)
      at Promise._settlePromiseFromHandler (http://localhost:4200/__cypress/static/js/vendor.js:44621:31)
      at Promise._settlePromiseAt (http://localhost:4200/__cypress/static/js/vendor.js:44697:18)
      at Promise._settlePromises (http://localhost:4200/__cypress/static/js/vendor.js:44813:14)
      at Async._drainQueue (http://localhost:4200/__cypress/static/js/vendor.js:42180:16)
      at Async._drainQueues (http://localhost:4200/__cypress/static/js/vendor.js:42190:10)
      at MutationObserver.Async.drainQueues (http://localhost:4200/__cypress/static/js/vendor.js:42018:14)

**Comment by mlunoe** [comment]
That is because of this error unrelated to this PR: https://github.com/dcos/dcos-ui/pull/1953

**Comment by mlunoe** [comment]
Rebased off of master where the fix has now merged

**Comment by MatApple** [comment]
Was just about to merge! Haha

**Comment by mlunoe** [comment]
Run integration tests

**Pull request merged**